### PR TITLE
feature: run rejection sampler in device tensor

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

1. make rejection sampler fully running in device tensor 
2. refactor the process of building compile option

## Issues
- add clamp func to remove negative numbers in `num_accepted`. (It's weird...)
-> It is resolved by forcing the dtype of target_probs to float16.
---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
4. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

**Comment by rebel-wonsubkim** [comment]
requires https://github.com/rebellions-sw/rebel_compiler/pull/11436

File: vllm_rbln/v1/sample/rbln_rejection_sampler.py [hunk 1]
```diff
@@ -162,13 +162,19 @@ def forward(
...
+        # (dlf16); feeding float32 makes the compiler reinterpret the f32 bits
+        # as dlf16 via a `dummy_cast` (a bit-level relabel, not a numeric
+        # conversion), which corrupts the probabilities so the NPU primitive
+        # returns a negative num_accepted. Casting here trades some rejection-
+        # sampling precision for correctness on the NPU.
         if sampling_metadata.all_greedy:
             # For greedy decoding, `target_logits` is already a one-hot tensor
             # where the max logit is set to 1 and the rest are set to 0.
-            target_probs = target_logits
+            target_probs = target_logits.to(torch.float16)
```
Comment (rebel-wonsubkim) [inline vllm_rbln/v1/sample/rbln_rejection_sampler.py:175]: rejection sampler input device tensor에 대한 type casting은 compiler 에서 해결되도록 수정 진행 중이니 제거하는게 맞을것 같습니다.

File: vllm_rbln/v1/sample/rbln_rejection_sampler.py [hunk 2]
```diff
@@ -162,13 +162,19 @@ def forward(
...
         else:
-            target_probs = target_logits.softmax(dim=-1, dtype=torch.float32)
-
+            target_probs = target_logits.softmax(dim=-1, dtype=torch.float16)
```
Comment (rebel-wonsubkim) [inline vllm_rbln/v1/sample/rbln_rejection_sampler.py:177]: 위와 마찬가지로 compiler 에서 자동적으로 device tensor value type casting 이 발생하는게 맞을것 같습니다.
type casting은 제거 하시죠

**Review by rebel-wonsubkim — CHANGES_REQUESTED** [review]
rejection sampler input device tensor value (!= reference) 에 대한 type casting 은 compiled graph 에서 처리되는게 맞아 보입니다.
host mode에서 weight sharing이 깨지는것 같은데 이것은 수정하지 않아도 될까요?
